src/spikeinterface/core/segmentutils.py

```diff
@@ -156,7 +156,7 @@ def __init__(self, parent_segments, sampling_frequency, ignore_times=True):
         BaseRecordingSegment.__init__(self, **time_kwargs)
         self.parent_segments = parent_segments
         self.all_length = [rec_seg.get_num_samples() for rec_seg in self.parent_segments]
-        self.cumsum_length = np.cumsum([0] + self.all_length)
+        self.cumsum_length = [0] + [sum(self.all_length[: i + 1]) for i in range(len(self.all_length))]
```

Comment:
Why list instead of numpy arrays ? Is it because of the type of scalar that are not python type ?

Reply:
Yes, this is used to determine start and stop frames below.

Reply:
Yes I know but why list is better that numpy arrays for this ?

Reply:
If you get an element from a numpy array (e.g. array[index] with type int of some sort) the resulting type is not an int but a scalar array int that can overflow. This avoids having scalar arrays.